api/python/ai/chronon/cli/tui/widgets.py

```
@@ -0,0 +1,573 @@
+        for i, (node, step_idx) in enumerate(self.steps):
+            if node == target_node and step_idx == target_step_idx:
+                return i
+        return None
+
+    def _move_vertically(self, direction: str):
+        """Move to next/previous node, maintaining step position."""
+        current_node, current_step_idx = self.steps[self.current_step_index]
+        current_node_index = self.node_list.index(current_node)
+        
+        node_range = (range(current_node_index + 1, len(self.node_list)) if direction == "down" 
+                     else range(current_node_index - 1, -1, -1))
```

Flake8 (7.2.0)

[error] continuation line under-indented for visual indent

(E128)